tests/container_test.go

```diff
@@ -804,3 +804,44 @@ func stackWithBool(t *testing.T, stack *vm.Stack, v bool) {
 	require.True(t, ok)
 	require.Equal(t, v, res)
 }
+
+func TestPutMeta(t *testing.T) {
+	c, cBal, _ := newContainerInvoker(t, false)
+
+	t.Run("meta disabled", func(t *testing.T) {
+		acc := c.NewAccount(t)
+		cnt := dummyContainer(acc)
+		balanceMint(t, cBal, acc, containerFee*1, []byte{})
+
+		metaInfo, err := stackitem.Serialize(stackitem.NewMapWithValue(
+			[]stackitem.MapElement{{Key: stackitem.Make("cid"), Value: stackitem.Make(cnt.id[:])}}))
+		require.NoError(t, err)
+
+		c.Invoke(t, stackitem.Null{}, "put", cnt.value, cnt.sig, cnt.pub, cnt.token, false)
+		c.InvokeFail(t, "container does not support meta-on-chain", "putObject", metaInfo, nil)
+
+		expected := stackitem.NewStruct([]stackitem.Item{
+			stackitem.NewByteArray(cnt.value),
+			stackitem.NewByteArray(cnt.sig),
+			stackitem.NewByteArray(cnt.pub),
+			stackitem.NewByteArray(cnt.token),
+		})
+		c.Invoke(t, expected, "get", cnt.id[:])
+	})
+
+	t.Run("meta enabled", func(t *testing.T) {
+		acc := c.NewAccount(t)
+		cnt := dummyContainer(acc)
+		balanceMint(t, cBal, acc, containerFee*1, []byte{})
+
+		metaInfo, err := stackitem.Serialize(stackitem.NewMapWithValue(
+			[]stackitem.MapElement{
+				{Key: stackitem.Make("cid"), Value: stackitem.Make(cnt.id[:])},
+				{Key: stackitem.Make("oid"), Value: stackitem.Make(cnt.id[:])},
+			}))
+		require.NoError(t, err)
+
+		c.Invoke(t, stackitem.Null{}, "put", cnt.value, cnt.sig, cnt.pub, cnt.token, true)
+		c.Invoke(t, stackitem.Null{}, "putObject", metaInfo, nil)
```

Comment:
if we have tools to test whether notification event is expectedly thrown, lets do it. If no, suggest to create an issue for this. Suppose other methods/contracts need this too

Reply:
we use neo-go's test framework. i tried my best, @roman-khimov, check if there is more optimal way to test events